File: src/main/java/com/matag/admin/auth/SecurityContextHolderHelper.java
package com.matag.admin.auth;

import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

import com.matag.admin.session.MatagSession;
import com.matag.admin.user.MatagUser;

@Component
public class SecurityContextHolderHelper {
  public MatagUser getUser() {
    return (MatagUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
  }

  public MatagSession getSession() {
    return (MatagSession) SecurityContextHolder.getContext().getAuthentication().getCredentials();
  }
}

File: src/main/java/com/matag/admin/auth/SecurityContextHolderHelper.kt
package com.matag.admin.auth

import com.matag.admin.session.MatagSession
import com.matag.admin.user.MatagUser
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
open class SecurityContextHolderHelper {
    fun getUser() =
        SecurityContextHolder.getContext().authentication?.principal as MatagUser

    fun getSession() =
        SecurityContextHolder.getContext().authentication?.credentials as MatagSession
}
